geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/executor/cluster/ClusterExecutor.java

```diff
@@ -145,28 +91,34 @@ private RedisResponse getSlots(ExecutionHandlerContext ctx) {
    * </pre>
    *
    * Note that there are no 'slave' entries since Geode does not host all secondary data apart from
-   * primary as redis does. The cluster port is provided only for consistency with the format
-   * of the output.
+   * primary as redis does. The cluster port is provided only for consistency with the format of the
+   * output.
    */
   private RedisResponse getNodes(ExecutionHandlerContext ctx) {
-    List<MemberBuckets> memberBuckets = getMemberBuckets(ctx);
     String memberId = ctx.getMemberName();
+    Map<String, List<Integer>> memberBuckets =
+        ctx.getRegionProvider().getSlotAdvisor().getMemberBuckets();
+    List<SlotAdvisor.MemberBucketSlot> memberBucketSlots =
+        ctx.getRegionProvider().getSlotAdvisor().getBucketSlots();
 
     StringBuilder response = new StringBuilder();
-    for (MemberBuckets m : memberBuckets) {
-      response.append(String.format("%s %s:%d@%d master",
-          m.getMemberId(), m.getHostAddress(), m.getPort(), m.getPort()));
+    for (Map.Entry<String, List<Integer>> member : memberBuckets.entrySet()) {
+      List<Integer> buckets = member.getValue();
+      SlotAdvisor.MemberBucketSlot mbs = memberBucketSlots.get(buckets.get(0));
```

Comment:
Something seems tangled here if we have to assume that buckets have at least 1 entry and we are always calling get(0).

Reply:
Actually it assumes that members (nodes) have at least one bucket. I'd prefer to have just one data structure that handles both slots:nodes and nodes:slots but I can't figure out how to avoid a bunch of redundancy.

Reply:
There's probably something fun we could do with streams Collectors.groupingBy to take a list an group it by slots or group it by nodes, but anyway this probably works.